utils/src/lib.rs

```diff
@@ -65,6 +161,23 @@ pub fn remove_extra_costs(gas: U256, calldata: &[u8]) -> U256 {
     gas.saturating_sub(calldata_cost.into()).saturating_sub(BASE_TX_COST.into())
 }
 
+pub fn next_nonce(
+    caller: Address,
+    provider_url: &str,
+    block: Option<BlockId>,
+) -> Result<U256, ProviderError> {
+    let provider = Provider::try_from(provider_url).expect("Bad fork_url provider");
+    let rt = RuntimeOrHandle::new();
+    rt.block_on(provider.get_transaction_count(caller, block))
+}
+
+pub fn next_create_address(caller: Address, nonce: U256) -> Address {
+    let mut stream = rlp::RlpStream::new_list(2);
+    stream.append(&caller);
+    stream.append(&nonce);
+    H256::from_slice(keccak256(&stream.out()).as_slice()).into()
+}
```

Comment:
Replace with this function from ethers https://github.com/gakonst/ethers-rs/blob/master/ethers-core/src/utils/mod.rs#L152